staging/src/kubevirt.io/api/core/v1/schema.go

```diff
@@ -598,6 +601,40 @@ type DiskTarget struct {
 	PciAddress string `json:"pciAddress,omitempty"`
 }
 
+type LaunchSecurity struct {
+	// AMD Secure Encrypted Virtualization (SEV).
+	SEV *SEV `json:"sev,omitempty"`
+}
+
+type SEV struct {
+	// Guest policy flags as defined in AMD SEV API specification.
+	// Note: due to security reasons it is not allowed to enable guest debugging. Therefore NoDebug flag is not exposed to users and is always true.
+	Policy *SEVPolicy `json:"policy,omitempty"`
+}
+
+type SEVPolicy struct {
```

Comment:
Well, mostly in theory :) Unfortunately the documentaiton is not very 'detailed'. From what I know:

- `NoDebug` - if set to `true` prevents debugging the guest memory with SEV ioctl cmds `KVM_SEV_DBG_DECRYPT`/`KVM_SEV_DBG_ENCRYPT`. In this PR it is not exposed in VMI spec and always enabled due to security reasons.
- `EncryptedState` - controls whether `SEV ES` should be enabled or not. This PR does not introduce 'full' support of the feature. It only allows setting this policy flags. There is still a need to detect `ES` capability on the node and label it appropriately. I will work on it in the follow-up PR.
- other policy flags AFAIK control the behaviour during snapshot/restore and migration flows (which are currently not supported)

One more thing is that the policy bitmask is used during the attestation and it is included in the measuremnt. So once it is set by a user it cannot be changed during the lifetime of the guest.

Reply:
Thank you for the explanation. Perhaps we didn't need to expose the `EncryptedState` right now or at least it would be good to have a webhook that rejects VMI specs that uses it. 
I'm not very worried about this as this feature is behind a feature gate and this PR is very large already. Perhaps you can follow up?

Reply:
Sure, I also plan to bring in SEV ES and SNP support. Agree that this PR is already big enough. As soon as the basic SEV gets in I will work on those features (+attestation) in the follow-up PRs.